[2024-07-13T02:32:29.800+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-13T02:32:29.810+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: model_training_v86.preprocess_data scheduled__2024-07-12T00:00:00+00:00 [queued]>
[2024-07-13T02:32:29.814+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: model_training_v86.preprocess_data scheduled__2024-07-12T00:00:00+00:00 [queued]>
[2024-07-13T02:32:29.815+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-07-13T02:32:29.823+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): preprocess_data> on 2024-07-12 00:00:00+00:00
[2024-07-13T02:32:29.829+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=1136) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-07-13T02:32:29.829+0000] {standard_task_runner.py:63} INFO - Started process 1157 to run task
[2024-07-13T02:32:29.830+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'model_training_v86', 'preprocess_data', 'scheduled__2024-07-12T00:00:00+00:00', '--job-id', '407', '--raw', '--subdir', 'DAGS_FOLDER/model_training.py', '--cfg-path', '/tmp/tmpu5cf2uvd']
[2024-07-13T02:32:29.831+0000] {standard_task_runner.py:91} INFO - Job 407: Subtask preprocess_data
[2024-07-13T02:32:29.864+0000] {task_command.py:426} INFO - Running <TaskInstance: model_training_v86.preprocess_data scheduled__2024-07-12T00:00:00+00:00 [running]> on host 0e0c69066946
[2024-07-13T02:32:29.926+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='model_training_v86' AIRFLOW_CTX_TASK_ID='preprocess_data' AIRFLOW_CTX_EXECUTION_DATE='2024-07-12T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-12T00:00:00+00:00'
[2024-07-13T02:32:29.927+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-13T02:32:30.341+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  драма
[2024-07-13T02:32:30.345+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  комедия
[2024-07-13T02:32:30.348+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  криминал
[2024-07-13T02:32:30.350+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  ужасы
[2024-07-13T02:32:30.353+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  фантастика
[2024-07-13T02:32:30.356+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  триллер
[2024-07-13T02:32:30.359+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  семейный
[2024-07-13T02:32:30.362+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  документальный
[2024-07-13T02:32:30.365+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  биография
[2024-07-13T02:32:30.367+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  боевик
[2024-07-13T02:32:30.370+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  короткометражка
[2024-07-13T02:32:30.373+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  детектив
[2024-07-13T02:32:30.376+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  история
[2024-07-13T02:32:30.379+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  мелодрама
[2024-07-13T02:32:30.382+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  приключения
[2024-07-13T02:32:30.385+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  музыка
[2024-07-13T02:32:30.388+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  военный
[2024-07-13T02:32:30.391+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  фэнтези
[2024-07-13T02:32:30.394+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  мюзикл
[2024-07-13T02:32:30.398+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  концерт
[2024-07-13T02:32:30.400+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  спорт
[2024-07-13T02:32:30.403+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  для взрослых
[2024-07-13T02:32:30.406+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  детский
[2024-07-13T02:32:30.410+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  реальное ТВ
[2024-07-13T02:32:30.413+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  церемония
[2024-07-13T02:32:30.416+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  вестерн
[2024-07-13T02:32:30.419+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  игра
[2024-07-13T02:32:30.421+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  новости
[2024-07-13T02:32:30.424+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  фильм-нуар
[2024-07-13T02:32:30.427+0000] {logging_mixin.py:188} INFO - Добавлен жанр:  ток-шоу
[2024-07-13T02:32:30.431+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-13T02:32:30.431+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/indexes/base.py", line 3791, in get_loc
    return self._engine.get_loc(casted_key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "index.pyx", line 152, in pandas._libs.index.IndexEngine.get_loc
  File "index.pyx", line 181, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 7080, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 7088, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'country'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/tasks/preprocess_data.py", line 160, in preprocess_data
    get_binary_country(country)
  File "/opt/airflow/dags/tasks/preprocess_data.py", line 153, in get_binary_country
    float_rows = data[data['country'].apply(lambda x: isinstance(x, float))]
                      ~~~~^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/frame.py", line 3893, in __getitem__
    indexer = self.columns.get_loc(key)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/indexes/base.py", line 3798, in get_loc
    raise KeyError(key) from err
KeyError: 'country'
[2024-07-13T02:32:30.441+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=model_training_v86, task_id=preprocess_data, run_id=scheduled__2024-07-12T00:00:00+00:00, execution_date=20240712T000000, start_date=20240713T023229, end_date=20240713T023230
[2024-07-13T02:32:30.449+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 407 for task preprocess_data ('country'; 1157)
[2024-07-13T02:32:30.484+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-13T02:32:30.497+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-13T02:32:30.500+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
